diff -ru yum-3.2.27-orig/docs/yum.conf.5 yum-3.2.27/docs/yum.conf.5
--- yum-3.2.27-orig/docs/yum.conf.5	2010-09-16 14:46:30.953767292 -0400
+++ yum-3.2.27/docs/yum.conf.5	2010-09-16 14:52:52.767615747 -0400
@@ -126,6 +126,8 @@
 List of packages that should only ever be installed, never updated. Kernels
 in particular fall into this category. Defaults to kernel, kernel-smp,
 kernel-bigmem, kernel-enterprise, kernel-debug, kernel-unsupported.
+Note that "kernel-modules" is \fBnot\fR in this list, in RHEL-6, and so anything
+providing that is updated like any other package.
 
 .IP
 \fBinstallonly_limit \fR
Only in yum-3.2.27/docs: yum.conf.5~
diff -ru yum-3.2.27-orig/yum/config.py yum-3.2.27/yum/config.py
--- yum-3.2.27-orig/yum/config.py	2010-09-16 14:46:30.989699062 -0400
+++ yum-3.2.27/yum/config.py	2010-09-16 14:50:29.539641005 -0400
@@ -626,7 +626,7 @@
     proxy_username = Option()
     proxy_password = Option()
     installonlypkgs = ListOption(['kernel', 'kernel-bigmem',
-            'kernel-enterprise','kernel-smp', 'kernel-modules', 'kernel-debug',
+            'kernel-enterprise','kernel-smp', 'kernel-debug',
             'kernel-unsupported', 'kernel-source', 'kernel-devel', 'kernel-PAE',
             'kernel-PAE-debug'])
     # NOTE: If you set this to 2, then because it keeps the current kernel it
